test/meson.build

```diff
@@ -0,0 +1,135 @@
+    'ta/ta_utils.c'
+]
+# The zimg code requires using threads. On windows, threads
+# also requires timer code so this is added.
+if features['win32-internal-pthreads']
+    test_utils_args += '-DWIN32_TESTS'
+    test_utils_files += ['osdep/timer.c',
+                         'osdep/timer-win2.c',
+                         'osdep/win32/pthread.c',
+                         'osdep/windows_utils.c']
+endif
+
+test_utils_deps = [libavutil, libm, pthreads]
+if features['win32-desktop']
+    test_utils_deps += cc.find_library('winmm')
+endif
+test_utils_objects = libmpv.extract_objects(test_utils_files)
+test_utils = static_library('test-utils', 'test_utils.c', include_directories: incdir,
+                            c_args: test_utils_args, objects: test_utils_objects,
+                            dependencies: test_utils_deps)
+
+# For getting imgfmts and stuff.
+img_utils_files = [
+    'misc/thread_pool.c',
+    'osdep/threads.c',
+    'video/csputils.c',
+    'video/fmt-conversion.c',
+    'video/img_format.c',
+    'video/mp_image.c',
+    'video/sws_utils.c'
+]
+if features['zimg']
+    img_utils_files += ['video/repack.c', 'video/zimg.c']
+endif
+
+img_utils_objects = libmpv.extract_objects(img_utils_files)
+img_utils = static_library('img-utils', 'img_utils.c', include_directories: incdir,
+                           dependencies: [libavcodec], objects: img_utils_objects)
+
+# The actual tests.
+chmap_files = [
+    'audio/chmap_sel.c'
+]
+if features['av-channel-layout']
+    chmap_files += 'audio/chmap_avchannel.c'
+endif
+chmap_objects = libmpv.extract_objects(chmap_files)
+chmap = executable('chmap', 'chmap.c', include_directories: incdir,
+                   objects: chmap_objects, link_with: test_utils)
+test('chmap', chmap)
+
+gl_video_objects = libmpv.extract_objects('video/out/gpu/ra.c',
+                                          'video/out/gpu/utils.c')
+gl_video = executable('gl-video', 'gl_video.c', objects: gl_video_objects,
+                      include_directories: incdir, link_with: [img_utils, test_utils])
+test('gl-video', gl_video)
+
+json = executable('json', 'json.c', include_directories: incdir, link_with: test_utils)
+test('json', json)
+
+linked_list = executable('linked-list', files('linked_list.c'), include_directories: incdir)
+test('linked-list', linked_list)
+
+paths_objects = libmpv.extract_objects('options/path.c', path_source)
+paths = executable('paths', 'paths.c', include_directories: incdir,
+                   objects: paths_objects, link_with: test_utils)
+test('paths', paths)
+
+if get_option('libmpv')
+    libmpv_test = executable('libmpv-test', 'libmpv_test.c',
+                             include_directories: incdir, link_with: libmpv)
+    test('libmpv', libmpv_test)
+endif
+
+# Minimum required libavutil version that works with these tests.
+# Will need to be manually updated when ffmpeg adds/removes more formats in the future.
+if libavutil.version().version_compare('>= 57.9.100')
```

Comment:
If this is for the pix_fmt tests, did a quick check and `p210` was added in FFmpeg/ffmpeg@b2cd1fb2ec64d04841d6c06fddf6e1d6e4ecfb34 , which is after FFmpeg/ffmpeg@b1f1de084416768e8b4ba76b338029b974e518a2 (57.9.101). The following bump was in FFmpeg/ffmpeg@b236ef0a594e20038b4045e2fecd414f1886d212 (57.10.101).

x2bgr10 was added in FFmpeg/ffmpeg@0760d9153c39e95e175c434e56916e7d950a4f03 (57.7.100), and finally the PSEUDOPAL flag removal that is noticed in some formats was done in FFmpeg/ffmpeg@985c0dac674846721ec8ff23344c16ac7d1c9a1e (57.0.100).

Thus, the overall highest version required seems to be **57.10.101**.

(with version 5.1 of FFmpeg the AVChannelLayout stuff got added but no new pix_fmts. The next pix_fmt which would cause a failure would be in the upcoming 6.0 release at FFmpeg/ffmpeg@6ab8a9d375ca922b2a94cd7160a4e3c5abe6339c , but that we can update as we hit that)